### PR TITLE
Fix memory leak and ARC GC issue 

### PR DESCRIPTION
**Pull request by oleksinski**
Fixed memory leak found during analyzing and profiling phases.
Fixed GC ARC issue casing application to crash: static variable reference was pointing to a temporary string object which was eligible for cleaning by Garbage Collector.

### CODE REVIEW

File: Pod/Classes/JNKeychain.m [hunk 1]
```diff
@@ -36,7 +36,7 @@ + (NSMutableDictionary *)getKeychainQuery:(NSString *)key forAccessGroup:(NSStri
         [keychainQuery setObject:[self getFullAppleIdentifier:group] forKey:(__bridge id)kSecAttrAccessGroup];
     }
     
-    return [keychainQuery mutableCopy];
```
Comment (oleksinski) [inline Pod/Classes/JNKeychain.m]: mutableCopy / copy methods always return a retained object which the caller is responsible for releasing.
When we’re creating an object using explicit allocation then we’re responsible for it and within the autorelease pool block it is automatically released when a callee doesn’t longer need an object.

File: Pod/Classes/JNKeychain.m [hunk 2]
```diff
@@ -125,31 +125,42 @@ + (id)loadValueForKey:(NSString *)key
...
-        OSStatus status = SecItemCopyMatching((__bridge CFDictionaryRef)query, (CFTypeRef *)&result);
-        if (status == errSecItemNotFound)
-        {
-            status = SecItemAdd((__bridge CFDictionaryRef)query, (CFTypeRef *)&result);
-        }
-        if (status == errSecSuccess)
-        {
-            NSString *accessGroup = [(__bridge NSDictionary *)result objectForKey:(__bridge NSString *)kSecAttrAccessGroup];
-            NSArray *components = [accessGroup componentsSeparatedByString:@"."];
-            bundleSeedID = [[components objectEnumerator] nextObject];
```
Comment (oleksinski) [inline Pod/Classes/JNKeychain.m]: Without making a copy of bundleSeedID we are referencing the value of an internal data structure which is eligible for releasing.

**Comment by jeremangnr** [comment]
good catch! one minor comment, could you change the if's so the opening brace is not on a new line? just so we are consistent with the other if clauses

**Comment by jeremangnr** [comment]
@oleksinski I also just noted in the Travis log (https://travis-ci.org/jeremangnr/JNKeychain/builds/92004108) that some tests are failing:
```
2015-11-19 09:48:55.121 JNKeychain[1132:60b] Running Keychain test with access group sharing
2015-11-19 09:48:55.134 JNKeychain[1132:60b] Failed to save! for access group 'demo.JNKeychain.shared'
2015-11-19 09:48:55.136 JNKeychain[1132:60b] Value for key 'accessGroupTestKey' is: '(null)' for access group 'demo.JNKeychain.shared'
2015-11-19 09:48:55.139 JNKeychain[1132:60b] Failed to delete! for access group 'demo.JNKeychain.shared'
```

**Comment by oleksinski** [comment]
@jeremangnr I will take a look

**Comment by oleksinski** [comment]
@jeremangnr The code formatting has changed and now conforms to an overall code formatting.
I've run a few more times tests locally and they have all passed successfully on my local environment.
I've also found that the very first travis build contains the same failure regarding access group tests execution.
Just find the key phrase "failure" within the first build and then correspondingly within the last one:
* https://travis-ci.org/jeremangnr/JNKeychain/builds/89449732
* https://travis-ci.org/jeremangnr/JNKeychain/builds/92345414

2015-11-20 21:10:26.100 JNKeychain[1133:60b] Cannot find executable for CFBundle 0x7b6c3610 </Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator7.1.sdk/System/Library/AccessibilityBundles/CertUIFramework.axbundle>

As far as I can see Travis CI uses iPhone Simulator v7.1 SDK and there is an outstanding issue with this iOS version number:
http://stackoverflow.com/questions/22310753/keychainwrapperitem-stopped-working-in-ios-7-1

Maybe this is the case.

**Comment by oleksinski** [comment]
And another one suggestion:
It would be nice to decrease the minimum iOS pod version required from iOS 7.0 to 6.0.
The library will also work fine with iOS 6.0 projects.
In our case we can't use JNKeychain as a pod dependency since projects are configured to have minimum target iOS version 6.0. This is the reason why projects might hold the library sources instead of having it as a pod dependency.

**Comment by oleksinski** [comment]
@jeremangnr, any news?

**Comment by jeremangnr** [comment]
hey @oleksinski, sorry for the delay. i've been super busy (you know how end of the year is) but i'll have a look at this before friday.

**Comment by oleksinski** [comment]
Thanx! And appreciate for lowering minimum iOS version to 6.0.
Now it's possible to use JNKeychain through cocoapods!

**Comment by jeremangnr** [comment]
thank you for the fixes/new features! :metal:

**Pull request merged**